v-next/hardhat/src/internal/cli/telemetry/sentry/init.ts

```diff
@@ -1,118 +1,149 @@
 
 import {
+  captureException,
   createStackParser,
   functionToStringIntegration,
-  getIntegrationsToSetup,
   initAndBind,
   linkedErrorsIntegration,
   nodeStackLineParser,
-  requestDataIntegration,
   ServerRuntimeClient,
   stackParserFromStackParserOptions,
 } from "@sentry/core";
+import debug from "debug";
 
-import { getHardhatVersion } from "../../../utils/package.js";
-
-import { onUncaughtExceptionIntegration } from "./integrations/onuncaughtexception.js";
-import { onUnhandledRejectionIntegration } from "./integrations/onunhandledrejection.js";
 import { nodeContextIntegration } from "./vendor/integrations/context.js";
 import { contextLinesIntegration } from "./vendor/integrations/contextlines.js";
 import { createGetModuleFromFilename } from "./vendor/utils/module.js";
 
-interface InitOptions {
+const log = debug("hardhat:core:sentry:init");
+
+interface GlobalCustomSentryReporterOptions {
+  /**
+   * Sentry's DSN
+   */
   dsn: string;
-  transport: (transportOptions: BaseTransportOptions) => Transport;
-  release?: string;
-  serverName?: string;
-  integrations?: (integrations: Integration[]) => Integration[];
+
+  /**
+   * The environment used to report the events
+   */
+  environment: string;
+
+  /**
+   * The release of Hardhat
+   */
+  release: string;
+
+  /**
+   * A transport that customizes how we send envelopes to Sentry's server.
+   *
+   * See the transport module for the different options.
+   */
+  transport: Transport;
 }
 
+const GENERIC_SERVER_NAME = "<user-server>";
+
 /**
- * Initialize Sentry for Node, without performance instrumentation.
+ * This function initializes a custom global sentry reporter/client.
+ *
+ * There are two reasons why we customize it, instead of using the default one
+ * provided by @sentry/node:
+ *   - @sentry/node has an astronomical amount of dependencies -- See https://github.com/getsentry/sentry-javascript/discussions/13846
+ *   - We customize the transport to avoid blocking the main Hardhat process
+ *     while reporting errors.
+ *
+ * Once you initialize the custom global sentry reporter, you can use the usual
+ * `captureException` and `captureMessage` functions exposed by @sentry/core.
+ *
+ * The reason that this uses the global instance of sentry (by calling
+ * initAndBind), is that using the client directly doesn't work with the linked
+ * errors integration.
+ *
+ * Calling `init` will also install the global unhandled rejection and uncaught
+ * exception handlers.
  */
-export async function init(options: InitOptions): Promise<void> {
-  const stackParser = stackParserFromStackParserOptions(
-    createStackParser(nodeStackLineParser(createGetModuleFromFilename())),
+export function init(options: GlobalCustomSentryReporterOptions): void {
+  initAndBind<ServerRuntimeClient, ServerRuntimeClientOptions>(
+    ServerRuntimeClient,
+    {
+      dsn: options.dsn,
+      environment: options.environment,
+      serverName: GENERIC_SERVER_NAME,
```

Comment:
Ah, maybe anonymization isn't needed!

Reply:
it wouldn't hurt being cautions. I changed it.

Reply:
Addressed